nexus/src/sagas.rs

```diff
@@ -866,17 +877,28 @@ async fn sim_instance_migrate(
 ) -> Result<(), ActionError> {
     let osagactx = sagactx.user_data();
     let params = sagactx.saga_params();
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
 
     let migration_id = sagactx.lookup::<Uuid>("migrate_id")?;
     let dst_sled_uuid = params.migrate_params.dst_sled_uuid;
     let dst_propolis_uuid = sagactx.lookup::<Uuid>("dst_propolis_id")?;
     let (instance_id, old_runtime) =
         sagactx.lookup::<(Uuid, InstanceRuntimeState)>("migrate_instance")?;
 
+    // Allocate an IP address the destination sled for the new Propolis server.
+    let propolis_addr = osagactx
+        .datastore()
+        .next_ipv6_address(&opctx, dst_sled_uuid)
+        .await
+        .map_err(ActionError::action_failed)?;
```

Comment:
Interesting, I hadn't considered that (obviously). It seemed like the allocation of a new address was harmless, but I agree that if we send the request to the sled agent _and then crash_, and send it again, the sled agent _could_ see that as a different request. I think the current implementation of the instance-ensure endpoint in the sled agent only looks at the ID of the instance to determine if it already has an existing instance, as opposed to the rest of the data.

But in any case, I agree with you. We may ultimately change that instance-ensure logic to check other properties -- in fact, we _know_ we'll do that, because otherwise users can't update the instance, for example, with new disks, NICs, or changing the amount of memory. Thanks for flagging this, I'll move it out to another saga action.

Reply:
Done in [66d1c52](https://github.com/oxidecomputer/omicron/pull/891/commits/66d1c5208c3c0467e9b3227a6cea82387a202c74)